### ISSUE TITLE
CLI --score-transpose flag ignored in Musescore 4

### ISSUE THREAD

**Issue by jongbinjung** [musescore/MuseScore#24723]
### Issue type

Import/export issue

### Description with steps to reproduce

Run 

```sh
mscore file.mscz --score-transpose '{"mode":"by_key","direction":"closest","targetKey":0,"transposeInterval":null,"transposeKeySignatures":true,"transposeChordNames":true,"useDoubleSharpsFlats":false}'
```

and parse the result (either `pdf` or `mscz`) and the resulting output is unchanged, regardless of   what `'targetKey'` is set to.

### Supporting files, videos and screenshots

[example.zip](https://github.com/user-attachments/files/17003703/example.zip) that includes an `input.mscz` in the key of C, and several pairs of `output-targetKey{key}.pdf` and `output-targetKey{key}.mscz` files that were generated via a simple `python` script:

```python 
import json
import base64
from pathlib import Path
import subprocess


def main():
    """Main function"""

    score = Path("input.mscz")
    for key in range(-7, 7):
        config = (
            f'{{"mode":"by_key","direction":"closest","targetKey":{key},'
            '"transposeInterval":null,"transposeKeySignatures":true,"transposeChordNames":true,'
            '"useDoubleSharpsFlats":false}'
        )
        a = subprocess.run(
            [
                "mscore",
                score.absolute(),
                "--score-transpose",
                config,
            ],
            capture_output=True,
        )
        base64_pdf = json.loads(a.stdout)["pdf"]
        base64_mscz = json.loads(a.stdout)["mscz"]
        Path(f"output-targetKey{key}.pdf").write_bytes(base64.b64decode(base64_pdf))
        Path(f"output-targetKey{key}.mscz").write_bytes(base64.b64decode(base64_mscz))


if __name__ == "__main__":
    main()
```

The `python` script is just to quickly try many different keys (all result in outputs that are unchanged from the original); but the problem persist even if run directly in a terminal. (e.g.,

```sh
mscore input.mscz \
        --score-transpose '{"mode":"by_key","direction":"closest","targetKey":7,"transposeInterval":null,"transposeKeySignatures":true,"transposeChordNames":true,"useDoubleSharpsFlats":false}' \
        | jq -r .pdf | base64 --decode | > output.pdf
```

### What is the latest version of MuseScore Studio where this issue is present?

4.4.0

### Regression

Yes, this used to work in MuseScore 3.x and now is broken

### Operating system

macOS 14.6.1

### Additional context

Also found a [related bug report](https://musescore.org/en/node/354550) at musescore.org. 

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by jongbinjung** [comment]
Forgot to add `stderr` output:

```
qt.qml.typeregistration: Invalid QML element name "IconCode"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "MusicalSymbolCodes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "ContainerType"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "NavigationEvent"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "MUAccessible"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "CompareType"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "SelectionMode"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "ToolBarItemType"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "DockToolBarAlignment"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "Location"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "CloudVisibility"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "SaveToCloudResponse"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "DirectionTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "SlurTieTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "NoteHead"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "Beam"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "Hairpin"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "OrnamentTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "Dynamic"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "Glissando"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "CommonTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "BarlineTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "MarkerTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "KeySignatureTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "AccidentalTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "FretDiagramTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "LineTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "TextTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "ArticulationTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "AmbitusTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "ChordSymbolTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "BendTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "TremoloBarTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "TremoloTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "VoiceTypes"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "InstrumentsTreeItemType"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "StandardButton"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "SaveLocationType"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "GenerateAudioTimePeriodType"; value type names should begin with a lowercase letter
qt.qml.typeregistration: Invalid QML element name "MigrationType"; value type names should begin with a lowercase letter
2024-09-14 13:51:08.263 mscore[78127:806874] WARNING: Secure coding is automatically enabled for restorable state! However, not on all supported macOS versions of this application. Opt-in to secure coding explicitly by implementing NSApplicationDelegate.applicationSupportsSecureRestorableState:.
```

**Comment by jongbinjung** [comment]
Tried to take a stab this weekend, but didn't get too far. So far, I've confirmed:

1. The command line eventually calls [BackendApi::applyTranspose](https://github.com/musescore/MuseScore/blob/master/src/converter/internal/compat/backendapi.cpp#L750), 
2. This in turn calls [NotationInteraction::transpose](https://github.com/musescore/MuseScore/blob/master/src/notation/internal/notationinteraction.cpp#L4380), which is also what seems to be called when a score is transposed via GUI
3. The command line version then calls [BackendApi::doExportScoreTranspose](https://github.com/musescore/MuseScore/blob/master/src/converter/internal/compat/backendapi.cpp#L617) with a `INotationPtr` to the supposedly `transpose`'d notation; but as far as I can tell, the earlier `BackendApi::applyTranspose` has had no effect on the notation?

I'm very new to the Musescore code base, and it's been almost a decade since I've dealt with `cpp`; so the way I confirmed that last part was to force-write an `mscz` file before/after `applyTranspose`:

```diff
+ Score* score = notation->elements()->msScore();
+ std::string pre_file = score->name().toStdString() + "_pre.mscz";
+ MscWriter::Params pre_params;
+ pre_params.mode = MscIoMode::Zip;
+ pre_params.filePath = QString::fromStdString(pre_file);
+ MscWriter pre_msczWriter(pre_params);
+ pre_msczWriter.open();
+ score->masterScore()->project().lock()->writeMscz(pre_msczWriter, false, true);
+ pre_msczWriter.close();

Ret ret = applyTranspose(notation, optionsJson);
if (!ret) {
    return ret;
}

+ std::string post_file = score->name().toStdString() + "_post.mscz";
+ MscWriter::Params post_params;
+ post_params.mode = MscIoMode::Zip;
+ post_params.filePath = QString::fromStdString(post_file);
+ MscWriter post_msczWriter(post_params);
+ post_msczWriter.open();
+ score->masterScore()->project().lock()->writeMscz(post_msczWriter, false, true);
+ post_msczWriter.close();
```

and then do a diff on the _internal `mscx`_ file fo the decompressed `mscz` archives; which indicated minimal difference:

```sh
$ diff pre.mscx post.mscx
5c5
<   <LastEID>1129</LastEID>
---
>   <LastEID>1130</LastEID>
``` 

Tried digging here and there, but pretty lost now. Any help would be much appreciated!

**Comment by jongbinjung** [comment]
Just doubled checked that this works with `3.6.2`, but not with `4.0 Alpha 2`; so definitely seems to have been broken with 4.0 release.

**Comment by JohnOmernik** [comment]
I am eagerly following this issue as I have this same problem!  It looks like a fix was proposed and tested, and on Nov 9 added to MuseScore Studio 4.4.4 but then removed on Nov 12. Any ideas when this could be moved to a release? I am on 4.4.4 and am still seeing this issue, and would LOVE to have this updated!  

Eagerly awaiting!

**Comment by cbjeukendrup** [comment]
The fix was merged to the master branch, and at some point we were considering porting it to 4.4.4, but we decided to limit 4.4.4 to urgent changes only. But the fix will be part of 4.5.

**Comment by JohnOmernik** [comment]
Awesome! Thank you for the quick response. Appreciate all the hard work that goes into this product!
